File: src/ZOMOdds.cpp
// Header files

#include "ZOMApp.h"
#include "ZOMOdds.h"

// Constructors

ZOMOddsSettings::ZOMOddsSettings()
{
  zomPer_ = 0.0;
  fightPer_ = 0.0;
  numZom_ = 0.0;
  return;
};

ZOMOddsSettings::ZOMOddsSettings(double zombie, double fight)
{
  zomPer_ = zombie;
  fightPer_ = fight;
  numZom_ = 0.0;
  return;
};

ZOMOddsSettings::~ZOMOddsSettings()
{
  return;
};

// Functions

void ZOMOddsSettings::runCalc()
{
  numZom_ = zomPer_;
  numZom_ /= (ZOM_MAX_PERCENT-zomPer_);
  numZom_ /= (fightPer_/ZOM_MAX_PERCENT);
  numZom_ = ceil(numZom_);
  return;
};

int ZOMOddsSettings::getNumZom()
{
  return ((int)numZom_);
};

void ZOMOddsSettings::printResults()
{
  // Print header
  cout << '\n';
  cout << "--- ZOMOdds test ---" << '\n';

  // Print settings
  cout << '\n';
  cout << "Settings" << '\n';
  cout << "--------" << '\n';
  cout << "  *Zombified humans: " << zomPer_;
  cout << "%" << '\n';
  cout << "  *Survivors in the fight : " << fightPer_;
  cout << "%" << '\n';
  cout << '\n';

  // Print results
  cout << "Results" << '\n';
  cout << "-------" << '\n';
  cout << "  *Number of zombies each survivor needs to eliminate: ";
  cout << ((int)numZom_) << '\n' << '\n';

  return;
};

File: src/ZOMOdds.kt

package src

class ZOMOdds() : Brains() {
    private var zomPer: Double = 0.0
    private var fightPer: Double = 0.0
    private var numZom: Double = 0.0

    constructor(zombie: Double, fight: Double) : this() {
        zomPer = zombie
        fightPer = fight
        numZom = 0.0
    }

    override fun promptForInput() {
        if (printToScreen) {
            println("Percent of the population turned:")
            zomPer = readln().toDouble()
            println("Percent of the population capable of fighting:")
            fightPer = readln().toDouble()
        }
    }

    private fun errorChecker() {
        var debugOutput = "Checking for errors..."
        if (zomPer > 100)
        {
            debugOutput = "$debugOutput\n  "
            debugOutput += "Warning: Population cannot be more than 100% zombies."
            zomPer = 100.0
        }
        if (zomPer < 0)
        {
            debugOutput = "$debugOutput\n  "
            debugOutput += "Warning: Population cannot be less than 0% zombies."
            zomPer = 0.0
        }
        if (fightPer > 100)
        {
            debugOutput = "$debugOutput\n  "
            debugOutput += "Warning: More than 100% of the humans fighting is impossible."
            fightPer = 100.0
        }
        if (fightPer < 0)
        {
            debugOutput = "$debugOutput\n  "
            debugOutput += "Warning: Less than 0% of the humans fighting is impossible."
            fightPer = 0.0
        }
        if (printToScreen) {
            println(debugOutput)
            println("")
        }
    }

    override fun runCalc() {
        if (printToScreen) {
            printHeader()
        }
        errorChecker()
        val maxPercent = 100.0
        numZom = zomPer
        numZom /= (maxPercent - zomPer)
        numZom /= (fightPer / maxPercent)
        numZom = kotlin.math.ceil(numZom)
    }

    private fun getNumZom(): Int {
        return numZom.toInt()
    }

    override fun printResults() {
        println("ZOMOdds Settings")
        println("--------")
        println("  * Zombified humans: $zomPer%")
        println("  * Survivors in the fight: $fightPer%")
        println("")

        println("Results")
        println("-------")
        println("  * Number of zombies each survivor needs to eliminate: " + getNumZom().toString())
    }
}
